MParT/RectifiedMultivariateExpansion.h

```diff
@@ -72,22 +64,15 @@ namespace mpart{
 
                     // Create a subview containing only the current point
                     auto pt = Kokkos::subview(pts, Kokkos::ALL(), ptInd);
-                    auto pt_off = Kokkos::subview(pt, Kokkos::pair<int,int>(0,pt.size()-1));
 
                     // Get a pointer to the shared memory that Kokkos set up for this team
                     Kokkos::View<double*,MemorySpace> cache(team_member.thread_scratch(1), cacheSize);
 
-                    // Fill in entries in the cache that are independent of x_d.  By passing DerivativeFlags::None, we are telling the expansion that no derivatives with wrt x_1,...x_{d-1} will be needed.
-                    worker_off.FillCache1(cache.data(), pt_off, DerivativeFlags::None);
-                    worker_off.FillCache2(cache.data(), pt_off, pt_off(pt_off.size()-1), DerivativeFlags::None);
-
-                    // Evaluate the expansion
-                    output_slice(ptInd) = worker_off.Evaluate(cache.data(), coeff_off);
 
                     // Fill in entries in the cache that are dependent on x_d.  By passing DerivativeFlags::None, we are telling the expansion that no derivatives with wrt x_1,...x_{d-1} will be needed.
-                    worker_diag.FillCache1(cache.data(), pt, DerivativeFlags::None);
-                    worker_diag.FillCache2(cache.data(), pt, pt(pt.size()-1), DerivativeFlags::None);
-                    output_slice(ptInd) += worker_diag.Evaluate(cache.data(), coeff_diag);
+                    worker.FillCache1(cache.data(), pt, DerivativeFlags::None);
+                    worker.FillCache2(cache.data(), pt, pt(pt.size()-1), DerivativeFlags::None);
+                    output_slice(ptInd) = worker.Evaluate(cache.data(), coeff);
```

Comment:
See my comment above about the form of this expansion.  I was expecting something very different.

Reply:
resolved?